jax/lax/lax.py

```diff
@@ -3123,13 +3164,21 @@ def _select_batch_rule(batched_args, batch_dims, **unused_kwargs):
     on_false = broadcast(on_false, pred.shape)
   return select(pred, on_true, on_false), 0
 
+def _select_masking_rule(padded_vals, logical_shapes):
+  pred_shape, true_shape, false_shape = map(lambda x: masking.padded_shape_as_value(x.shape), padded_vals)
+  assert onp.array_equal(pred_shape, true_shape)
+  assert onp.array_equal(pred_shape, false_shape)
+  # ensure all arguments have the same, concrete shape:
+  return select(*(reshape(val, pred_shape) for val in padded_vals))
```

Comment:
It seems the rming the reshape has no effect. I wonder if it only mattered for situations which are now unsupported (particularly since there is now no reshape masking rule).

Reply:
(Resolved)